### ISSUE TITLE
changeTab() not changes when route params changed

### ISSUE THREAD

**Issue by mxurshid** [rpocklin/ui-router-tabs#12]
I think there is some small bug in following line:
https://github.com/rpocklin/ui-router-tabs/blob/master/src/ui-router-tabs.js#L88

In my case, for all tab items 'route' are the same, 'params' are different.
For example:
```javascript 
$scope.tabData = [
      {
          heading: 'Settings 1',
          route:   'user.settings',
          params: { id: 1 }
      },
      {
          heading: 'Settings 2',
          route:   'user.settings',
          params: { id: 2 }
      }
];
```

As you can see, changeTab() not changes.

**Comment by rpocklin** [comment]
Yep you are right, it's a bug introduced with the commit 673f408477585ebe333da217bd45d14c32ac3713.  I'll look into fixing it, thanks for highlighting it!

**Comment by rpocklin** [comment]
Can you please checkout the branch `better-state-equality-test` and tell me if this fixes your issue?

**Comment by mxurshid** [comment]
Yes, It works now for route params! But when I navigate from browser <code>ui-view</code> shows tab-content, but tab not automatically selected. I'm not sure, maybe it is related with my code. I am testing it.
Thanks!

**Comment by rpocklin** [comment]
Try it now - the tab should appear automatically as before - let me know if it doesn't.

**Comment by mxurshid** [comment]
Now It works!
Thank you!

**Comment by rpocklin** [comment]
Cool i'll look at putting this into a release, thanks for testing.

**Comment by rpocklin** [comment]
Landed with `1.3.0`.
